### ISSUE TITLE
🐞 Bug – Frequent core-dumps due to lots of reloading

### ISSUE THREAD

**Issue by silky** [elm-land/elm-land#162]
## Problem

I'm getting frequent crashes of `elm-land server` due to file-watching reloading.

Here's the error:

```
[391253:0x36d88a0]   595994 ms: Mark-sweep 2010.2 (2092.3) -> 2000.1 (2095.8) MB, 2231.9 / 0.0 ms  (average mu = 0.079, current mu = 0.014) task; scavenge might not succeed
[391253:0x36d88a0]   598106 ms: Mark-sweep 2004.5 (2095.8) -> 2001.3 (2097.3) MB, 2096.1 / 0.0 ms  (average mu = 0.042, current mu = 0.008) allocation failure; GC in old space requested


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
 1: 0xab4728 node::Abort() [node]
 2: 0x9a4839  [node]
 3: 0xcc23e0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
 4: 0xcc27ab v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
 5: 0xea6cc5  [node]
 6: 0xebb109 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [node]
 7: 0xe9804a v8::internal::HeapAllocator::AllocateRawWithLightRetrySlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
 8: 0xe98c44 v8::internal::HeapAllocator::AllocateRawWithRetryOrFailSlowPath(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [node]
 9: 0xe7934d v8::internal::Factory::NewFillerObject(int, v8::internal::AllocationAlignment, v8::internal::AllocationType, v8::internal::AllocationOrigin) [node]
10: 0x1261892 v8::internal::Runtime_AllocateInOldGeneration(int, unsigned long*, v8::internal::Isolate*) [node]
11: 0x16abdf9  [node]
zsh: IOT instruction (core dumped)  npx elm-land server
```

The reloading is occuring because I'm generating a static css file; elm-land detects this an reloads, but it does it about 5-10 times before failing in this way.

## Your setup

> - What operating system (Mac/Window/Linux) are you using? Please provide a version!

NixOS

> - Please share the output of the following commands:
>    - `node -v`
>    - `elm-land -v`

```
> npx node -v
v18.18.2
> npx elm-land -v

🌈  Elm Land (v0.19.5) is currently installed.
```

**Comment by silky** [comment]
Fow what it's worth it's significantly worse in the latest version of elm-land; it crashes almost every time I regenerate the CSS.

**Comment by silky** [comment]
Interestingly, it seems like some node/filesystem bug.

If I run

```
echo css/Css.hs | entr -cs 'runghc css/Css.hs>./static/css.css'
```

it causes the coredump; but if I change the `entr` command to:

```
echo css/Css.hs | entr -cs 'runghc css/Css.hs>/tmp/css.css && cp /tmp/css.css ./static/css.css'
```

it works fine.

So I suppose this is almost certainly some nodejs issue.

**Comment by ryan-haskell** [comment]
Good news– I just released [📦 Elm Land v0.20.0](https://github.com/elm-land/elm-land/releases/tag/v0.20.0) which resolves this issue (more details in the release linked above)

Thanks for improving Elm Land! 🌈
